### ISSUE TITLE
[NES]: Unknown operator S in Gimmik(U) and Mr.Gimmik(EU)

### ISSUE THREAD

**Issue by retro-wertz** [meleu/RetroArch-problematic-cheevos#29]
Permission to hijack this repo, since i dont know of a place for retroachievement issues for retroarch :D

Mr.Gimmik and Gimmik will show **.[ERROR] [CHEEVOS]: Unknown operator S.** multiple times. Not sure if this can be replicated in standalone ranesas I dont have windows installed.

**Comment by leiradel** [comment]
Thanks for the report, this looks like an issue with the cheevos implementation in RetroArch. Can you add a link to the games' pages in Retro Achievements?

**Comment by retro-wertz** [comment]
here: http://retroachievements.org/Game/2139

**Comment by meleu** [comment]
I confirm that. Here is the `grep CHEEVO` of the log:

```txt
[INFO] [CHEEVOS]: system RAM: 0x7ffb2cad2b60 2048
[INFO] [CHEEVOS]: save RAM:   (nil) 0
[INFO] [CHEEVOS]: video RAM:  (nil) 0
[INFO] [CHEEVOS]: RTC:        (nil) 0
[INFO] [CHEEVOS]: testing NES (discards VROM).
[INFO] [CHEEVOS]: getting game id for hash 6da79d16e353c7bb9f5e6dbcd63d81f3
[INFO] [CHEEVOS]: got game id 2139.
[INFO] [CHEEVOS]: got achievements for game id 2139.
[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[ERROR] [CHEEVOS]: Unknown operator S
.[INFO] [CHEEVOS]: deactivated unlocked achievements in softcore mode.
[INFO] [CHEEVOS]: deactivated unlocked achievements in hardcore mode.
[INFO] [CHEEVOS]: posted playing activity.
[INFO] [CHEEVOS]: posted playing activity.
[INFO] [CHEEVOS]: awarding cheevo 4025: Health Potion (Obtain a Health Restore Potion ).
[INFO] [CHEEVOS]: awarded achievement 4025.
[INFO] [CHEEVOS]: awarding cheevo 4026: Pink Fireball (Obtain a Pink Fireball).
[INFO] [CHEEVOS]: awarded achievement 4026.
[INFO] [CHEEVOS]: awarding cheevo 4027: Bomb (Obtain a Bomb ).
[INFO] [CHEEVOS]: awarded achievement 4027.
```

**Comment by GameDragon2k** [comment]
The **S** element represents an Alt Group. In the case of Gimmick, there are 5 achievements that use alt groups.

http://retroachievements.org/Achievement/4025
http://retroachievements.org/Achievement/4026
http://retroachievements.org/Achievement/4027
http://retroachievements.org/Achievement/27625
http://retroachievements.org/Achievement/29649

**Comment by leiradel** [comment]
I'm working on this right now, can anyone provide me with a savestate just before one of the cheevos listed by @GameDragon2k while I debug the parser? Any NES core compatible with RetroAchievements will do. Thanks.

**Comment by leiradel** [comment]
Hm, I can't retro it. The log doesn't show any parsing error messages, nor a breakpoint in the line where the message is generated hits.

```
$ ./retroarch -v -L ../../RetroArch/cores/quicknes_libretro.dll ../../RetroArch/content/Gimmick\!\ \(Japan\).nes 2>&1 |grep CHEEVOS [INFO] [CHEEVOS]: system RAM: 0000000006DA6300 2048
[INFO] [CHEEVOS]: save RAM:   0000000006D71A50 8192
[INFO] [CHEEVOS]: video RAM:  0000000000000000 0
[INFO] [CHEEVOS]: RTC:        0000000000000000 0
[INFO] [CHEEVOS]: testing NES (discards VROM).
[INFO] [CHEEVOS]: getting game id for hash 9faa5d8c9f1060d3e6296ae50c63a6ca
[INFO] [CHEEVOS]: got game id 2139.
[INFO] [CHEEVOS]: posted playing activity.
[INFO] [CHEEVOS]: posted playing activity.
```

**Comment by retro-wertz** [comment]
i do not see the Unknown operator S anymore, was that an event?

**Comment by leiradel** [comment]
Maybe there was a glitch somewhere, I dunno. Are the cheevos listed by @GameDragon2k working?

**Comment by retro-wertz** [comment]
verifying, but i think the cheevos is inconsistent. (pink fireball)

**Comment by retro-wertz** [comment]
[Gimmick! (Japan)(PinkFireball).state.tar.gz](https://github.com/meleu/RetroArch-problematic-cheevos/files/1404080/Gimmick.Japan.PinkFireball.state.tar.gz)
 Here is pink fireball cheevos. not sure why, but you need to pick that "thing" twice. pick it once, then die and pick it again.

**Comment by leiradel** [comment]
Is the behavior the same in the official emulator?

---- On Sat, 21 Oct 2017 13:21:20 +0100 notifications@github.com wrote ----

Gimmick! (Japan)(PinkFireball).state.tar.gz
Here is pink fireball cheevos. not sure why, but you need to pick that "thing" twice. pick it once, then die and pick it again.

**Comment by retro-wertz** [comment]
not using windows atm, so not sure.

**Comment by meleu** [comment]
I'm not getting those messages anymore.

I talked with @GameDragon2k and he said that he reuploaded those cheevos that use the Alt group.

Curiously this was enough to "fix" those messages.

@retro-wertz I'm closing this issue, but feel free to reopen if you feel that the problem isn't solved on your end.

**Comment by leiradel** [comment]
I'd like to confirm if what @retro-wertz said about having to pick up the pink fireball twice to trigger the cheevo also happens in the official emulator though. Maybe it should stay opened until then?

**Comment by meleu** [comment]
I got that pink fireball with no issues. Got only once and it was enough to trigger. I'll test it again and if it happens I reopen this.

**Comment by meleu** [comment]
Jumping to conclusions: @leiradel no need to worry about RetroArch incompatibilities with this game. ;-)

Looks like some Gimmick cheevos are a little buggy...

I said in my previous post that I got the Pink Fireball only once and it works, but I died on the first enemies (I mean, I got the item only once, but it was after loosing one life).

I've made more tests and when I got the Pink Fireball the cheevo didn't pop, and then I died on purpose and got the item again. This time it triggered.

Another test: I got the Pink Fireball without losing a life and the cheevo didn't pop, but when I pressed `down` in the d-pad (to "equip" the item) the cheevo worked.

**This behavior happens on RetroArch AND RANes**.

I'll try to report it to the respective cheevos developer.
